hydroflow_cli_integration/src/lib.rs

```diff
@@ -393,3 +448,44 @@ where
         self.sink.take().unwrap()
     }
 }
+
+#[pin_project]
+pub struct MergeSource<T, S: Stream<Item = T> + Send + Sync + ?Sized> {
+    marker: PhantomData<T>,
+    #[pin]
+    sources: Vec<Pin<Box<S>>>,
+}
```

Comment:
`pin_project` seems unnecessary here? Already have `Pin<Box<S>>`

Reply:
Worth noting there is also `StreamExt::merge()` https://docs.rs/tokio-stream/latest/tokio_stream/trait.StreamExt.html#method.merge and `StreamMap` https://docs.rs/tokio-stream/latest/tokio_stream/struct.StreamMap.html in tokio